### PR TITLE
Add doxygen doc generation.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This integrates the doxygen docs generated by the engine in PR https://github.com/flutter/engine/pull/43915 into the API docs site using dummy Dart packages, and adds extraction of the various sections from the engine-uploaded documentation artifacts.

## Related Issues
 - https://github.com/flutter/flutter/issues/130999
 - https://github.com/flutter/flutter/issues/124833

### CODE REVIEW

**Comment by justinmc** [comment]
Cool idea!

**Comment by gspencergoog** [comment]
Thanks, but not my idea. I'm just adding the missing platform docs to it: we already have this for iOS and Android.

File: dev/docs/README.md [hunk 1]
```diff
@@ -14,6 +14,14 @@ This site hosts Flutter's API documentation. Other documentation can be found at
 the following locations:
 
 * [flutter.dev](https://flutter.dev) (main site)
+* [api.flutter.dev](https://api.flutter.dev) (API docs reference site)
+* Engine Embedder API documentation:
+  * [Android Embedder](https://api.flutter.dev/javadoc/index.html)
+  * [iOS Embedder](https://api.flutter.dev/ios-embedder/index.html)
```
Comment (christopherfujino) [inline dev/docs/README.md:20]: I get a 404 for this link

File: dev/docs/README.md [hunk 2]
```diff
@@ -14,6 +14,14 @@ This site hosts Flutter's API documentation. Other documentation can be found at
...
+  * [macOS Embedder](https://api.flutter.dev/macos-embedder/index.html)
```
Comment (christopherfujino) [inline dev/docs/README.md:21]: also 404

File: dev/docs/README.md [hunk 3]
```diff
@@ -14,6 +14,14 @@ This site hosts Flutter's API documentation. Other documentation can be found at
...
+  * [Linux Embedder](https://api.flutter.dev/linux-embedder/index.html)
```
Comment (christopherfujino) [inline dev/docs/README.md:22]: 404

File: dev/docs/README.md (hunk 1 above)
Comment (gspencergoog) [reply dev/docs/README.md:20]: Yes, because this PR adds these.

Comment (gspencergoog) [reply dev/docs/README.md:20]: When this PR is committed, docs generation will create them.

Comment (christopherfujino) [reply dev/docs/README.md:20]: :+1:

File: dev/tools/create_api_docs.dart [hunk 4]
```diff
@@ -23,6 +23,58 @@ import 'dartdoc_checker.dart';
...
+  'macos': PlatformDocsSection(
+    zipName: 'macos-docs.zip',
+    sectionName: 'macOS',
+    checkFile: 'interface_flutter_view.html',
+    subdir: 'macos-embedder',
+  ),
+  'linux': PlatformDocsSection(
+    zipName: 'linux-docs.zip',
+    sectionName: 'Linux',
+    checkFile: 'struct___fl_view.html',
```
Comment (christopherfujino) [inline dev/tools/create_api_docs.dart:61]: is this supposed to have 3 underscores in a row?

Comment (gspencergoog) [reply dev/tools/create_api_docs.dart:61]: Yep.  That's how Doxygen generates it.

Comment (christopherfujino) [reply dev/tools/create_api_docs.dart:61]: :+1:

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**